src/button-group/internal.tsx

```diff
@@ -95,7 +92,10 @@ const InternalButtonGroup = forwardRef(
         return;
       }
       // Ignore navigation when the focused element is not an item.
-      if (document.activeElement && !document.activeElement.matches(`.${testUtilStyles.item}`)) {
+      if (
+        document.activeElement &&
+        !('dataset' in document.activeElement && (document.activeElement as HTMLElement).dataset.itemid)
```

Comment:
Should we add a dev warning if finding a focusable item w/o itemid? That should only be possible if the consumers misuse the API, forgetting the item IDs.